public/app/plugins/panel/canvas/CanvasContextMenu.tsx

```diff
@@ -207,25 +208,12 @@ export const CanvasContextMenu = ({ scene, panel }: Props) => {
     }
   };
 
-  const contextMenuAction = (actionType: string) => {
+  const contextMenuAction = (actionType: LayerActionID) => {
     scene.selection.pipe(first()).subscribe((currentSelectedElements) => {
-      const currentSelectedElement = currentSelectedElements[0];
-      const currentLayer = currentSelectedElement.parent!;
-
-      switch (actionType) {
-        case LayerActionID.Delete:
-          currentLayer.doAction(LayerActionID.Delete, currentSelectedElement);
-          break;
-        case LayerActionID.Duplicate:
-          currentLayer.doAction(LayerActionID.Duplicate, currentSelectedElement);
-          break;
-        case LayerActionID.MoveTop:
-          currentLayer.doAction(LayerActionID.MoveTop, currentSelectedElement);
-          break;
-        case LayerActionID.MoveBottom:
-          currentLayer.doAction(LayerActionID.MoveBottom, currentSelectedElement);
-          break;
-      }
+      const currentLayer = currentSelectedElements[0].parent!;
+      currentSelectedElements.forEach((currentSelectedElement: ElementState) => {
+        currentLayer.doAction(actionType, currentSelectedElement);
+      });
```

Comment:
Minor thing, but do you think it would be possible / make sense to update selection state to newly duplicated elements after duplicate? Would help user move newly duplicated elements to a new location in bulk easier

Reply:
Yes, I've also noticed that while testing, totally makes sense. Looking into it!

Reply:
@nmarrs Looking at it I think that's also the case for the newly created items and not sure if for other cases too, I'll test it out. But I think it would make sense to make the changes in a separate issue to keep this PR more focused. What do you think?

Reply:
Yeah I think that makes sense, I'll approve this PR 👍🏻